tools/cldr-code/src/main/java/org/unicode/cldr/util/Units.java

```diff
@@ -81,6 +82,32 @@ public static String combinePattern(
         LONG_TO_SHORT = ImmutableBiMap.copyOf(longToShort);
     }
 
+    public static class TypeAndCore {
+        public String type;
+        public String core;
+    }
+    /**
+     * Returns the type and core for a unit, be it long or short
+     *
+     * @param longOrShortUnit
+     * @param core
+     * @return
+     * @return
+     */
+    public static TypeAndCore splitUnit(String longOrShortUnit, TypeAndCore typeAndCore) {
```

Comment:
is splitUnit part of determining the order of items, as in the xml? if so, ideally the order could depend less on special-purpose java methods that may be tightly coupled to many other parts of the code -- something simpler and more data-driven, and loosely coupled, might be more robust and maintainable

Reply:
We have unfortunately both long ids and short ids for units: length-kilometer and kilometer. The former are in our locale files, and used in the ST ordering, the latter are "real" and used in the code.
There is a ticket to drop the "length-" etc subtag, but for now we have to work with it.